### PR TITLE
Fix 4204

### LINKED ISSUE #4204: setErrorHandler is not called on plugin error
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.5.0

### Plugin version

4.2.0

### Node.js version

16.17.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

22.04

### Description

Hi,
When I try to return an error from fastify plugin, I see that setErrorHandler is not called. 

I don't know why because I call my error handler plugin before to register my new plugin.

Thanks for the help ! 

### Steps to Reproduce

- Go to https://github.com/flexper/fastify-auth-prisma
- run `pnpm dlx ts-node test/core/real.ts`

### Expected Behavior

Should return formated error from https://github.com/flexper/unify-fastify

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #4204 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/404s.test.js [hunk 1]
```diff
@@ -1944,3 +1944,68 @@ test('Send 404 when frameworkError calls reply.callNotFound', t => {
 
   t.end()
 })
+
+test('hooks are applied to not found handlers', async ({ equal }) => {
```
Comment (Uzlopak) [inline test/404s.test.js:1948]:
```suggestion
test('hooks are applied to not found handlers /1', async ({ equal }) => {
```

**Review by Uzlopak — APPROVED** [review]

**Review by jsumners — APPROVED** [review]

**Review by climba03003 — APPROVED** [review]
LGTM.

**Comment by climba03003** [comment]
I really think we should tidy up the code afterward (e.g. refactor, or comment for code flow). 
It is not straight forward to see the error and hard to debug.

**Comment by mcollina** [comment]
I agree!

**Pull request merged**